common/persistence/sql/sqlplugin/queue_messate.go

```diff
@@ -38,20 +45,35 @@ type (
 		MessagePayload []byte
 	}
 
+	// QueueMessagesFilter
+	QueueMessagesFilter struct {
+		QueueType persistence.QueueType
+		MessageID int64
+	}
+
+	// QueueMessagesRangeFilter
+	QueueMessagesRangeFilter struct {
+		QueueType    persistence.QueueType
+		MinMessageID int64
+		MaxMessageID int64
+		PageSize     int
+	}
+
 	// QueueMetadataRow represents a row in queue_metadata table
 	QueueMetadataRow struct {
 		QueueType persistence.QueueType
 		Data      []byte
 	}
 
-	Queue interface {
-		InsertIntoQueue(row *QueueRow) (sql.Result, error)
+	QueueMessage interface {
+		InsertIntoMessages(row []QueueRow) (sql.Result, error)
+		SelectFromMessages(filter QueueMessagesFilter) ([]QueueRow, error)
+		RangeSelectFromMessages(filter QueueMessagesRangeFilter) ([]QueueRow, error)
+		DeleteFromMessages(filter QueueMessagesFilter) (sql.Result, error)
+		RangeDeleteFromMessages(filter QueueMessagesRangeFilter) (sql.Result, error)
```

Comment:
Consider:
```suggestion
		Enqueue(row []QueueRow) (sql.Result, error)
		Select(filter QueueMessagesFilter) ([]QueueRow, error)
		SelectRange(filter QueueMessagesRangeFilter) ([]QueueRow, error)
		Delete(filter QueueMessagesFilter) (sql.Result, error)
		DeleteRange(filter QueueMessagesRangeFilter) (sql.Result, error)
```

Reply:
sample current convention: 
https://github.com/temporalio/temporal/blob/9d5c94e/common/persistence/sql/sqlplugin/history_transfer_tasks.go#L54-L64

plus the function name should explicitly express the functionality.

Reply:
Seems to me `FromTransferTasks` is useless there too.

Reply:
I usually run mental test and answer to the question "Does this particular word really adds more context?" If interface name is `QueueMessage` and method accepts `QueueMessagesFilter`, is it clear what `Select` method does? Seems to me it is.

Reply:
in my mind:
```
store1.Select
store2.Select
```
vs
```
store1.SelectFromMessage
store2.SelectFromTransferTasks
```
the latter one is more easier to read, plus the latter is the naming convention.

Reply:
BTW, we have a #810 task before making the entire persistence layer modular

Reply:
Plus the name / interface collision here: https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/interfaces.go#L41